shell/common/shell.h

```diff
@@ -391,6 +391,8 @@ class Shell final : public PlatformView::Delegate,
                      >
       service_protocol_handlers_;
   bool is_setup_ = false;
+  std::future<bool> async_waiting_subsystems_;
+  std::queue<std::future<void>> pending_tasks_before_setup_;
```

Comment:
You are editing and reading this queue from multiple threads, but it's not a threadsafe datastructure.

Reply:
`pending_tasks_before_setup_` is only access on platform thread, it does not need to be thread safe.

Reply:
I read through the code and I think you are right.  It would be nice to enforce this.